### ISSUE TITLE
Fingerprint template/signature standard

### ISSUE THREAD

**Issue by mlinaje** [DanBrezeanu/R502-fingerprint#6]
Hi,
We are looking for fingerprint modules to build an open-source distributed heterogeneous (not always from the same vendor) to avoid network of locks (where each lock will be controlled by an MCU with WiFi). It will be the BsC thesis of one of my students. Summarizing, we are looking for fingerprint modules allowing to transfer the fingerprint from one lock (after registering a new user) to the rest of the locks (avoiding one registration per lock).

We found information about some modules, including R502, from HZGrow and your GitHub repository. Our main problem with this sensor is that we are not sure about the template/signature format they use for the fingerprint (to ensure compatibility with other fingerprint sensors to avoid vendor lock-in).

Thus, we need a fingerprint module compliance of the fingerprint template/signature format with some standards (ANSI, ISO, IAFIS,...) and that is the information that we do not find in the manuals provided. Do you have any experience/information about it?
Thanks in advance,

**Comment by DanBrezeanu** [comment]
Hello,
[Here](https://www.dropbox.com/sh/epucei8lmoz7xpp/AAAmon04b1DiSOeh1q4nAhzAa?dl=0) you can access the manuals for the available fingerprint modules from this manufacturer. This repository is designed and tested for the R502.

Regarding the format of the fingerprint it doesn't seem that it is following any standards (page 6 describes how a fingerprint image is built). If you need any real data from the module let me know and I will try to provide all the necessary files that you might need.

**Comment by mlinaje** [comment]
Thank you for the information @DanBrezeanu !
I have checked that the dropbox files you linked and they are the same provided by the company that I found on aliexpress.
I don't know yet if a fingerprint template downloaded could lead us to know more about the codification, but if we think we can figure it out from it, I will come back to you ;-) 
I have been searching a lot last weeks unluckily so finally I have asked some people (as you) for clues. Quite interesting but enigmatic fingerprint module. Also, I haven't found the algorithms used internally for the image pre-processing and matching. I doubt about the safety of a module with so much opacity.
Best,
